src/commands/executeFunction/eventGrid/OpenEventGridFileStep.ts

```diff
@@ -0,0 +1,85 @@
+import { AzExtFsExtra, AzureWizardExecuteStep, nonNullProp } from "@microsoft/vscode-azext-utils";
+import * as os from 'os';
+import * as path from "path";
+import * as vscode from 'vscode';
+import { type Progress } from "vscode";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../localize";
+import { feedUtils } from "../../../utils/feedUtils";
+import { type ExecuteEventGridFunctionContext } from "./ExecuteEventGridFunctionContext";
+
+export class OpenEventGridFileStep extends AzureWizardExecuteStep<ExecuteEventGridFunctionContext> {
+    public priority: number;
+
+    public async execute(context: ExecuteEventGridFunctionContext, progress: Progress<{ message?: string | undefined; increment?: number | undefined; }>): Promise<void> {
+        if (!context.eventSource || !context.selectedFileName || !context.selectedFileUrl) {
+            throw new Error('Internal error: event source or event type is missing');
+        }
+
+        // Get selected contents of sample request
+        const downloadingMsg: string = localize('downloadingSample', 'Downloading sample request...');
+        progress.report({ message: downloadingMsg });
+        context.selectedFileContent = await feedUtils.getJsonFeed(context, context.selectedFileUrl);
```

Comment:
You probably don't need to set this on the context since it's only used in this file.

Context values are typically reserved for values that get used across multiple steps.

You could then get rid of it from the typing.